src/index.ts

```diff
@@ -501,7 +501,14 @@ class DefaultVisualizer {
         if (getStructures) {
             copy.structures = [] as Structure[];
             for (let i = 0; i < this._dataset.structures.length; i++) {
-                copy.structures.push(this.structure.loadStructure(i, this._dataset.structures[i]));
+                const structure = this.structure.loadStructure(i, this._dataset.structures[i]);
+                if (structure instanceof Promise) {
+                    throw new Error(
+                        'loadStructure returned a Promise, but this code path must be synchronous'
+                    );
```

Comment:
It'd be helpful if the message explained why it could happen...

Reply:
I will turn this into an assert, as this is more to check that the underlying assumption we are making that this does never get called from an async path is not broken.